swarm/src/connection.rs

```diff
@@ -751,13 +764,30 @@ mod tests {
         ))
     }
 
+    #[test]
+    fn test_idle_timeout() {
```

Comment:
To actually test this feature fully, this test should wait for the configured idle-timeout, poll again and we should receive a keep-alive timeout error.

I'd suggest to change it to a `#[tokio::test]` for that and use `future::poll_fn` instead of `poll_noop_waker` to make sure we properly register the waker.